documentation/commands/extended_attributes/xattr_encoding.md

```diff
@@ -0,0 +1,31 @@
+###Extended Attributes (XATTR) Encoding
+
+If a Document has XATTRs and `DCP_OPEN_INCLUDE_XATTRS` flag was set as
+part of the `DCP_OPEN` message, then the `DCP_MUTATION` and
+`DCP_DELETION` messages will indicate the presence of the XATTRs by
+setting the new XATTR bit in the datatype field:
+
+    #define PROTOCOL_BINARY_DATATYPE_XATTR uint8_t(4)
+
+When this bit is set the first uint32_t in the body contains the
+size of the entire XATTR section.
+
+      Byte/     0       |       1       |       2       |       3       |
+         /              |               |               |               |
+        |0 1 2 3 4 5 6 7|0 1 2 3 4 5 6 7|0 1 2 3 4 5 6 7|0 1 2 3 4 5 6 7|
+        +---------------+---------------+---------------+---------------+
+       0| Total xattr length in network byte order                      |
+        +---------------+---------------+---------------+---------------+
+
+Following the length you'll find an iovector-style encoding of all of
+the XATTR key-value pairs with the following encoding:
+
+    uint32_t   length of next xattr pair: N + M + 1 (network order)
```

Comment:
I've not looked at the details in this document (and I think it would be better if it linked directly to: https://github.com/couchbase/memcached/blob/master/docs/Document.md#xattr---extended-attributes )